src/Orleans/Configuration/ClusterConfiguration.cs

```diff
@@ -419,58 +419,55 @@ public string ToString(string siloName)
 
         internal static async Task<IPAddress> ResolveIPAddress(string addrOrHost, byte[] subnet, AddressFamily family)
         {
-            var loopback = (family == AddressFamily.InterNetwork) ? IPAddress.Loopback : IPAddress.IPv6Loopback;
+            var loopback = family == AddressFamily.InterNetwork ? IPAddress.Loopback : IPAddress.IPv6Loopback;
+
+            // IF the address is an empty string, default to the local machine
+            if (string.IsNullOrEmpty(addrOrHost))
+            {
+                addrOrHost = Dns.GetHostName();
+            }
 
-            if (addrOrHost.Equals("loopback", StringComparison.OrdinalIgnoreCase) ||
-                addrOrHost.Equals("localhost", StringComparison.OrdinalIgnoreCase) ||
-                addrOrHost.Equals("127.0.0.1", StringComparison.OrdinalIgnoreCase))
+            // Fix StreamFilteringTests_SMS tests
+            if (addrOrHost.Equals("loopback", StringComparison.OrdinalIgnoreCase))
```

Comment:
I apologize for not being clear the first time I commented.

What I was getting at was that we specifically handled certain cases, and I wanted to ensure that we continued to do so.
Those cases were:
0.0.0.0 - which returned IPAddress.Any, and is currently handled by the IPAddress.TryParse.
127.0.0.1 - which returned loopback, and is currently handled by the IPAddress.TryParse.
"loopback" - which returned loopback, and is currently doing the same, with the last commit.
"localhost" - which returned loopback, and is not being handled as previously.

I understand that I'm being a bit of a stickler here, but we've many services using this, so we need to ensure we're not breaking any existing users without good cause.